### PR TITLE
Repository name correction, part3: test and sample data

### PR DESCRIPTION
**Pull request by awwad**
Moves sample and test data from directories like oem_metadata to directories like image_repo_metadata.

Adjusts test modules to find these files in the correct places.

Moves some test data into directories that will make more sense for them shortly.

### CODE REVIEW

**Comment by vladimir-v-diaz** [comment]
Do you want me to review this pull request?  Please assign reviewers so that we can keep track of pending requests.

**Comment by awwad** [comment]
I was still testing it. It's all good, so I'll add you now. (:

File: tests/test_data/pinned.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
   "delegations": [
     {
       "paths": ["*"],
-      "repositories": ["mainrepo", "director"]
+      "repositories": ["imagerepo", "director"]
```
Comment (vladimir-v-diaz) [inline tests/test_data/pinned.json:13]: Is there a reason why `imagerepo` is sometimes used, and `image_repo` at other times?

Comment (awwad) [reply tests/test_data/pinned.json:13]: Not a great one, no. Normally, if it's on its own, I just write 'imagerepo', and if it's part of some phrase in which words are already separated (like TEST_IMAGE_REPO_ROOT_FNAME), I separate the words.

There's an exception I can think of - there's an add_target_to_imagerepo call in the demo. That should probably be separated.

Comment (vladimir-v-diaz) [reply tests/test_data/pinned.json:13]: I feel that it should be `image_repo` everywhere.  "Image repository" is two words, and it's a bit harder to parse "imagerepo"

Comment (awwad) [reply tests/test_data/pinned.json:13]: Voila, an issue: #101

**Comment by vladimir-v-diaz** [comment]
You can go ahead and merge!

**Pull request merged**